pkg/odo/cli/component/ui/ui.go

```diff
@@ -218,6 +219,22 @@ func createComponentNameValidator(context *genericclioptions.Context) survey.Val
 	}
 }
 
+// Exists checks whether a component with the given name exists in the current application or not
+// componentName is the component name to perform check for
+// The first returned parameter is a bool indicating if a component with the given name already exists or not
+// The second returned parameter is the error that might occurs while execution
+func exists(client *occlient.Client, componentName, applicationName string) (bool, error) {
+	deploymentName, err := util.NamespaceOpenShiftObject(componentName, applicationName)
+	if err != nil {
+		return false, errors.Wrapf(err, "unable to create namespaced name")
+	}
+	deployment, _ := client.GetDeploymentConfigFromName(deploymentName)
```

Comment:
Why are we using occlient directly instead of `component.Exists()`?

Reply:
addressed